crates/subspace-networking/src/node.rs

```diff
@@ -386,7 +486,10 @@ impl Node {
             .await?;
 
         // TODO: A wrapper that'll immediately cancel query on drop
-        Ok(result_receiver)
+        Ok(WrapperWithPermits::new(
+            result_receiver,
+            vec![kademlia_permit, global_permit],
```

Comment:
The order of the drop is not guaranteed for Vector: https://doc.rust-lang.org/std/vec/struct.Vec.html
Some combination of calls and permit numbers could end up in hard-to-debug deadlocks.